# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/picture_graph.rs
# Commit: 8c70bc92ad36
# Full Hash: 8c70bc92ad36beb61f2cd6031e80ace3793fe53b
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-02-24 03:51:41
# Regressor Bug: 1749380
# File Overlap Count: 1
# Description:
#   Bug 1749380 - Part 1 - Improve how WR handles bounding rects for off-screen surfaces r=gfx-reviewers
#   
#   This patch introduces a number of subtle but important changes
#   to how we deal with off-screen surfaces. The overall goals are:
#   
# ==============================================================================

diff -r c6f1243c3ba4 -r 8c70bc92ad36 gfx/wr/webrender/src/picture_graph.rs
--- a/gfx/wr/webrender/src/picture_graph.rs	Wed Feb 23 20:42:19 2022 +0000
+++ b/gfx/wr/webrender/src/picture_graph.rs	Wed Feb 23 20:49:26 2022 +0000
@@ -106,12 +106,18 @@
 
                 let info = &mut self.pic_info[pic_index.0];
 
-                info.surface_index = Some(pictures[pic_index.0].assign_surface(
+                match pictures[pic_index.0].assign_surface(
                     frame_context,
                     tile_caches,
-                    parent_surface_index,
                     surfaces,
-                ));
+                ) {
+                    Some(surface_index) => {
+                        info.surface_index = Some(surface_index);
+                    }
+                    None => {
+                        info.surface_index = Some(parent_surface_index.unwrap());
+                    }
+                }
             }
         }
     }